### PR TITLE
Various fixes triggered by trying to get rid of uncovered code

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: lib/OpenQA/Scheduler.pm [hunk 1]
```diff
@@ -269,10 +269,6 @@ sub job_create {
     my ($settings, $no_notify) = @_;
     my %settings = %$settings;
 
-    if (my $error = OpenQA::Variables->new()->check(%settings)) {
```
Comment (aaannz) [inline lib/OpenQA/Scheduler.pm]: I can't find any other usage of Variables::check except this one. Why not remove the Variables.pm completely?

**Comment by coolo** [comment]
hmm, that was actually my intention. Must have got lost - but yes, Variables.pm is dead

**Comment by coolo** [comment]
removed it now

**Comment by coolo** [comment]
now even passing tests :)

**Pull request merged**